### PR TITLE
Document D4M Kubernetes support

### PR DESCRIPTION
**Pull request by mistyhacks**
cc/ @gbarr01 who can take this over since I'm sick.
This takes over from #5346 and supersedes it.

cc/ @gtardif @justincormack @vdemeester @johndmulhausen @friism

### CODE REVIEW

File: docker-for-mac/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+You can deploy a stack to Kubernetes using the `docker stack deploy` command,
+supplying a `docker-compose.yml` file.
+
+```bash
+$ docker stack deploy --compose-file /path/to/docker-compose.yml
+```
+
+You can see the service deployed using the `kubectl get services` command.
+
```
Comment (silvin-lubecki) [inline docker-for-mac/kubernetes.md]: Or maybe
```
docker stack services mystack
```
?

**Comment by mistyhacks** [comment]
See https://deploy-preview-5439--docker-docs.netlify.com/docker-for-mac/#kubernetes for how to enable, and https://deploy-preview-5439--docker-docs.netlify.com/docker-for-mac/#kubernetes for how to use it.

File: docker-for-mac/kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+### Specify a namespace
+
+By default, the `default` namespace is used. You can specify a namespace using
+the `--namespace` flag.
+
+```bash
+$ docker stack deploy --namespace my-app --compose-file /path/to/docker-compose.yml
```
Comment (silvin-lubecki) [inline docker-for-mac/kubernetes.md]: The stack name is missing here.

File: docker-for-mac/kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+While you are testing Kubernetes, you may want to deploy some workloads in swarm
+mode. You can use the `DOCKER_ORCHESTRATOR` variable to override the default
+orchestrator for a given terminal session or a single Docker command. The
+variable can be unset (the default, in which case Kubernetes is the
+orchestrator) or set to `swarm` or `kubernetes`. The following command
+overrides the orchestrator for a single deployment, by setting the variable at
+the start of the command itself.
+
+```bash
+DOCKER_ORCHESTRATOR=swarm docker stack deploy --compose-file /path/to/docker-compose.yml
```
Comment (silvin-lubecki) [inline docker-for-mac/kubernetes.md]: The stack name is missing here.

**Comment by mistyhacks** [comment]
@gbarr01 can you cherry-pick your commit into this PR? If you don't have my fork added as a remote, add it, then fetch and  check out the branch:

```
git remote add misty https://github.com/mistyhacks/docker.github.io
git fetch misty
git checkout d4m-k8s
git cherry-pick <SHA of your commit>
```

File: docker-for-mac/kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+  db:
+    build: db
+    image: dockerdemos/lab-db
+```
+
+If you already have a Kubernetes YAML file, you can deploy it using the
+`kubectl` command.
+
+TODO create a Kube YAML file to make the same app as the docker-compse.yml
```
Comment (mistyhacks) [inline docker-for-mac/kubernetes.md]: Don't forget to remove the TODO. :)

**Pull request merged**